Importing babel/standalone results in compilation error with Webpack

## Bug Report

**Current behavior**

Simply including `babel/standalone` results in an error `Module not found: Error: Can't resolve 'module' in 'project/node_modules/@babel/standalone'`

Repo to reproduce: https://github.com/matyasf/babel-standalone-bug

**Input Code**

clone the repo, run `yarn` and then `yarn run bundle` The code that is causing error during compilation:
```
import { transform } from '@babel/standalone'
const tt = transform
```

The error is:
```
ERROR in ./node_modules/@babel/standalone/babel.js 92843:15-32
Module not found: Error: Can't resolve 'module' in '/Users/matyas.szabo/CODE/instructure-ui/node_modules/@babel/standalone'
resolve 'module' in '/Users/matyas.szabo/CODE/instructure-ui/node_modules/@babel/standalone'
  Parsed request is a module
  using description file: /Users/matyas.szabo/CODE/instructure-ui/node_modules/@babel/standalone/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
...
```

**Expected behavior**

no error

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**

- Filename: `babel.config.js`: none

Webpack.config.js:

```js
const path = require('path')
const outputPath = path.resolve(__dirname, '__build__')
console.info(`Building documentation`)

module.exports = {
  mode: "development",
  module: {
    rules: [
      {
        test: /\.m?js$/,
        exclude: [/node_modules/, /\/lib\//, /\/es\//, /\.examples\.js$/],
        use: [
          {
            loader: 'babel-loader'
          }
        ]
      }
    ]
  },
  entry: {
    globals: './globals.js'
  },
  output: {
    path: outputPath,
    filename: '[name].js'
  }
}
```

**Environment**
<!--- Tip: Instead of filling out the questions below, you can run `npx envinfo --preset babel` and paste the result below ``` -->
```
  System:
    OS: macOS 10.15.7
  Binaries:
    Node: 15.2.1 - ~/.nvm/versions/node/v15.2.1/bin/node
    Yarn: 1.22.10 - ~/CODE/instructure-ui/node_modules/.bin/yarn
    npm: 7.5.3 - ~/.nvm/versions/node/v15.2.1/bin/npm
  npmPackages:
    @babel/core: ^7.13.1 => 7.13.1 
    @babel/standalone: ^7.13.7 => 7.13.7 
    babel-loader: ^8.2.2 => 8.2.2 
    webpack: ^5.24.2 => 5.24.2 
```

**Additional context**
<!-- Add any other context about the problem here. Or a screenshot if applicable -->

--- a/babel.config.js
+++ b/babel.config.js
@@ -76,7 +76,15 @@ module.exports = function (api) {
     case "standalone":
       includeRegeneratorRuntime = true;
       unambiguousSources.push("packages/babel-runtime/regenerator");
-    // fall through
+      convertESM = false;
+      ignoreLib = false;
+      // rollup-commonjs will converts node_modules to ESM
+      unambiguousSources.push(
+        "/**/node_modules",
+        "packages/babel-preset-env/data",
+        "packages/babel-compat-data"
+      );
+      break;
     case "rollup":
       convertESM = false;
       ignoreLib = false;
@@ -86,7 +94,7 @@ module.exports = function (api) {
         "packages/babel-preset-env/data",
         "packages/babel-compat-data"
       );
-      if (env === "rollup") targets = { node: nodeVersion };
+      targets = { node: nodeVersion };
       needsPolyfillsForOldNode = true;
       break;
     case "test-legacy": // In test-legacy environment, we build babel on latest node but test on minimum supported legacy versions

--- a/babel.config.js
+++ b/babel.config.js
@@ -75,14 +75,14 @@ module.exports = function (api) {
     // Configs used during bundling builds.
     case "standalone":
       includeRegeneratorRuntime = true;
-      unambiguousSources.push("packages/babel-runtime/regenerator");
       convertESM = false;
       ignoreLib = false;
       // rollup-commonjs will converts node_modules to ESM
       unambiguousSources.push(
         "/**/node_modules",
         "packages/babel-preset-env/data",
-        "packages/babel-compat-data"
+        "packages/babel-compat-data",
+        "packages/babel-runtime/regenerator"
       );
       break;
     case "rollup":

